cachito/web/models.py

```diff
@@ -286,6 +287,7 @@ class Request(db.Model):
     """A Cachito user request."""
 
     id = db.Column(db.Integer, primary_key=True)
+    created = db.Column(db.Date(), nullable=False, index=True, default=date.today())
```

Comment:
What was the reason to change this field from `datetime` to `date`? 

I believe this could be more useful as a datetime field, since we could use it for other purposes besides the date filter, and then having the exact creation timestamp would help.

Reply:
Here is the email that led me to this decision:

I believe the latter, all datetime values truncated to their date for the purpose of the comparison would be the most appropriate answer here.  Trying to make a given value more precise for comparison purposes I don't think is good behavior.  

-mike


On Wed, Sep 29, 2021 at 1:02 AM Daniel Cho <dacho@redhat.com> wrote:
- CLOUDBLD-6469
I have essentially completed this ticket. The first PR is ready to be approved and merged. The second PR will be ready to be approved and merged as soon as the first one is merged.

There is one question I have for the team, especially Michael:
For an example, the code I have implemented takes this - /api/v1/requests?start_date=2021-01-01&end_date=2021-01-01 - and treats the input not as 2021-01-01 but 2021-01-01T00:00:00.000000 instead. In other words, if the user intends to filter all Cachito requests within a single day, the end_date parameter would have to act as the exclusive upper limit (i.e. /api/v1/requests?start_date=2021-01-01&end_date=2021-01-02 for requests created on 1/1/2021).

The question is, should the code remain this way, or should the parameters be redesigned to only filter to the nearest date (i.e. all created date values would be truncated to their date for the purpose of this functionality)? The former would provide more control over the parameters, but the latter may be much easier to use.

Reply:
I see. Maybe we could handle the same behavior while keeping the field as `datetime`?

For example, we could set the `start_date` parameter to always behave as if had the `00:00:00.00000` timestamp and the `end_date` if it had the `23.59:59.99999` timestamp.

Wdyt?

Reply:
+1. Also, can the params be called `before` and `after` instead of `start_date`, `end_date`? Currently it sounds like the query should return only requests that started on day X and ended on day Y.

Reply:
yes please keep datetime

Reply:
I changed all of the `date` objects back to `datetime` objects. I've adjusted the `start_date` and `end_date` parameters just as you suggested, @brunoapimentel  (https://github.com/release-engineering/cachito/pull/570).

As for the name of the variables, @chmeliik, I agree with what you said. However, I want to suggest a new idea - `from_date` and `to_date`. I think this makes more sense, especially in the case where the two variables are equal to each other. I think that there is merit in keeping the word "date" in the name, since these variables represent `date`, not `datetime`. What do you think?

Reply:
`from_date`, `to_date` works for me.

Do the variables need to represent only `date` though? Could the endpoint accept both e.g. `/requests?from_date=2021-10-04` and `/requests?from_date=2021-10-04T20:00` and behave as expected? It may not even need extra logic in code, SQLAlchemy should be pretty flexible with datetime comparisons.

Reply:
I've changed the variables to `from_date` and `to_date`. I also added the `datetime` functionality for the parameters as well. You can check it out in this PR https://github.com/release-engineering/cachito/pull/570